sql/core/src/main/scala/org/apache/spark/sql/internal/SharedState.scala

```diff
@@ -191,11 +194,26 @@ private[sql] class SharedState(
 }
 
 object SharedState extends Logging {
-  try {
-    URL.setURLStreamHandlerFactory(new FsUrlStreamHandlerFactory())
-  } catch {
-    case e: Error =>
-      logWarning("URL.setURLStreamHandlerFactory failed to set FsUrlStreamHandlerFactory")
+  @volatile private var factory: Option[FsUrlStreamHandlerFactory] = None
+  private lazy val defaultFactory = new FsUrlStreamHandlerFactory()
```

Comment:
Do you need to hold a reference to this?

Reply:
We can skip this since `try .. catch` will work in the same way.

Reply:
I simplified this PR.